### PR TITLE
Remove the dependency to activesupport

### PR DESCRIPTION
**Pull request by mame**
... by replacing `demodulize` and `underscore` with hand-written regexps.

### CODE REVIEW

File: templates/template.rb [hunk 1]
```diff
@@ -48,12 +45,12 @@ class Type
 
       def initialize(yaml)
         @ruby_full_name = yaml["name"]
-        @ruby_class_name = @ruby_full_name.demodulize
+        @ruby_class_name = @ruby_full_name[/[^:]+\z/] # demodulize-like
+        name = @ruby_full_name.gsub("::", "_")
+        name = name.gsub(/(^)?(_)?([A-Z](?:[A-Z]*(?=[A-Z_])|[a-z0-9]*))/) { ($1 || $2 || "_") + $3.downcase } # underscore-like
         @c_function_name = if @ruby_full_name =~ /^RBS::Types/
```
Comment (ParadoxV5) [inline templates/template.rb:51]:
```suggestion
        @c_function_name = if @ruby_full_name.start_with? 'RBS::Types'
```

**Review by ParadoxV5 — COMMENTED** [review]
Fixes https://github.com/ruby/rbs/pull/2105#issuecomment-2507896809

`underscore`-like is so complex 😮.

**Comment by amomchilov** [comment]
`activesupport` was already an indirect dependency of RBS, and it still is after this PR, because `steep` depends on it. Is there a plan to remove `activesupport` from `steep`?

**Comment by HoneyryderChuck** [comment]
An alternative implementation is to provide the same method via a refinement.

**Comment by soutaro** [comment]
@amomchilov Yeah, it's subtle thing... The problem is for Ruby core developers who don't install full-development setup of RBS. They often develop without bundler or maybe rubygems, so having less dependency to minimum setup for RBS makes more sense for the case.

It was my fault that I didn't notice the problem when I was reviewing your initial PR. Looks like this change works good.

**Pull request merged**